boot/zephyr/main.c

```diff
@@ -506,20 +506,11 @@ void main(void)
 
     mcuboot_status_change(MCUBOOT_STATUS_STARTUP);
 
-#if (!defined(CONFIG_XTENSA) && DT_HAS_CHOSEN(zephyr_flash_controller))
-    if (!flash_device_get_binding(DT_LABEL(DT_CHOSEN(zephyr_flash_controller)))) {
-        BOOT_LOG_ERR("Flash device %s not found",
-		     DT_LABEL(DT_CHOSEN(zephyr_flash_controller)));
+    if (!flash_device_get_binding()) {
+        BOOT_LOG_ERR("Flash device not ready");
         while (1)
             ;
     }
```

Comment:
Do we even need this part?
How does the rest of booting process depend on this?
Is it only for checking whether there is device to boot from? If yes, can we do that at compile time?

The boot process will have flash_area_open invocation somewhere below that point, that will do the same check anyway, to get access to partition to boot.

Reply:
I'm no expert here, I'm happy to remove the code if those that work on mcuboot say its not needed.

Reply:
> I'm no expert here, I'm happy to remove the code if those that work on mcuboot say its not needed.

@nvlsianpu @utzig  Can you provide advise here?

Reply:
I think flash_device_get_binding() and its usage can be dropped. MCUboot will fail later on first attempt to the flash anyway. Since device is provide directly for DT-2-C API checking of that dosne't make bi sense anymoe.

Reply:
will update to remove `flash_device_get_binding`